tests/e2e/taqueria-cli-scaffolding-functional.spec.ts

```diff
@@ -1,38 +1,79 @@
-import { exec as exec1 } from 'child_process';
-import fsPromises from 'fs/promises';
-import util from 'util';
-const exec = util.promisify(exec1);
-
-const scaffoldDirName = `scrap/auto-test-taco-shop-functional`;
+import { prepareEnvironment } from '@gmrchk/cli-testing-library';
 
 describe('E2E Testing for taqueria scaffolding initialization,', () => {
-	beforeAll(async () => {
-		await fsPromises.rm(`${scaffoldDirName}`, { recursive: true, force: true });
-		await exec(`taq scaffold https://github.com/ecadlabs/taqueria-scaffold-taco-shop.git ${scaffoldDirName}`);
-	});
-
-	test('Verify that scaffold project is set up after running setup', async () => {
-		const appContents = await exec(`ls ${scaffoldDirName}/app`);
-		const taqContents = await exec(`ls ${scaffoldDirName}`);
-		expect(appContents.stdout).toContain('node_modules');
-		expect(taqContents.stdout).toContain('node_modules');
-		expect(taqContents.stdout).toContain('contracts');
-		expect(taqContents.stdout).toContain('artifacts');
-	});
-	test('Verify that scaffold project compiles contracts as part of setup', async () => {
-		const taqContents = await exec(`ls ${scaffoldDirName}/artifacts`);
-		expect(taqContents.stdout).toContain('hello-tacos.tz');
+	test('Verify that scaffold project sets up compiles contracts as part of setup', async () => {
+		const { execute, exists } = await prepareEnvironment();
+		const { code } = await execute(
+			'taq',
+			'scaffold https://github.com/ecadlabs/taqueria-scaffold-taco-shop.git auto-test-taco-shop-functional',
+		);
+		expect(code).toBe(0);
+		await exists('./auto-test-taco-shop-functional}/app');
```

Comment:
There is an extra right bracket after functional. It should be removed?

Btw we could probably make a variable because we are repeating `./auto-test-taco-shop-functional` too many times.

Reply:
Fixed.  don't need a variable. this is a test, its fine  we want it as easy to read as possible,